tasking/reaper.go

```diff
@@ -0,0 +1,325 @@
+package tasking
+
+import (
+	"context"
+	liberr "github.com/konveyor/controller/pkg/error"
+	"github.com/konveyor/tackle2-hub/model"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"os"
+	"path"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+const (
+	ReaperUnit = time.Hour
+)
+
+//
+// Reaper interface.
+type Reaper interface {
+	Run()
+}
+
+//
+// TaskReaper reaps tasks.
+type TaskReaper struct {
+	// DB
+	DB *gorm.DB
+	// k8s client.
+	Client client.Client
+}
+
+//
+// Run Executes the reaper.
+func (r *TaskReaper) Run() {
+	Log.Info("Reaping tasks.")
+	list := []model.Task{}
+	result := r.DB.Find(
+		&list,
+		"state IN ?",
+		[]string{
+			Created,
+			Succeeded,
+			Failed,
+		})
+	Log.Trace(result.Error)
+	if result.Error != nil {
+		return
+	}
+	for i := range list {
+		task := &list[i]
+		deleted, err := r.delete(task)
+		if err != nil {
+			Log.Trace(err)
+			continue
+		}
+		if deleted {
+			continue
+		}
+		err = r.emptyBucket(task)
+		if err != nil {
+			Log.Trace(err)
+			continue
+		}
+		err = r.deletePod(task)
+		if err != nil {
+			Log.Trace(err)
+			continue
+		}
+	}
+}
+
+//
+// delete Deletes the task as needed.
+func (r *TaskReaper) delete(task *model.Task) (deleted bool, err error) {
```

Comment:
This appears to always return `deleted=false`

Reply:
Fixed.